src/Components/Aspire.OpenAI/AspireOpenAIClientBuilderChatClientExtensions.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Data.Common;
+using Aspire.OpenAI;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using OpenAI;
+
+namespace Microsoft.Extensions.Hosting;
+
+/// <summary>
+/// Provides extension methods for registering <see cref="IChatClient"/> as a singleton in the services provided by the <see cref="IHostApplicationBuilder"/>.
+/// </summary>
+public static class AspireOpenAIClientBuilderChatClientExtensions
+{
+    private const string DeploymentKey = "Deployment";
+    private const string ModelKey = "Model";
```

Comment:
Why are we reading both `Deployment` and `Model` keys? Should we just pick one and go with it?

Reply:
Because OpenAI prefers "model" terminology while Azure OpenAI prefers "deployment". The intent is that developers don't need to worry about getting the wrong one, because at any given time they can specify either but not both. If we picked a single one, it may seem weird in cases where it doesn't match the backend you're using.

This isn't something I feel very strongly about, and we could force people to pick the key that corresponds to the backend they're using. It just seems like adding an extra pit of failure for no reason.

Reply:
I get confused every time I work on my related PR. Thanks to the explanation.

Reply:
Maybe at least stick this commentary into a code comment for the next person so they understand why both names are used.

Reply:
Done